Explorer/Assets/DCL/PerformanceAndDiagnostics/Diagnostics/ReportsHandling/Handlers/ReportHandlerBase.cs

```diff
@@ -59,29 +62,44 @@ public void LogException(Exception exception, ReportData reportData, Object cont
 
         internal abstract void LogExceptionInternal<T>(T ecsSystemException) where T: Exception, IDecentralandException;
 
-        internal abstract void LogExceptionInternal(Exception exception, ReportData reportData, Object context);
+        internal abstract void LogExceptionInternal(Exception exception, ReportData reportData, Object? context);
 
-        internal virtual void HandleSupressedException(Exception exception, ReportData reportData) { }
+        internal virtual void HandleSuppressedException(Exception exception, ReportData reportData) { }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [HideInCallstack]
         private bool IsLogMessageAllowed(in object message, in ReportData reportData, LogType logType) =>
             matrix.IsEnabled(reportData.Category, logType) && !Debounce(in message, in reportData, logType);
 
+        private bool IsLogMessageAllowed(in Exception message, in ReportData reportData, LogType logType) =>
+            matrix.IsEnabled(reportData.Category, logType) && !Debounce(in message, in reportData, logType);
+
+        private bool Debounce(in Exception message, in ReportData reportData, LogType logType)
+        {
+            if (!debounceEnabled) return false;
+
+            // Don't call DebounceInternal as exceptions can't be compared
+
+            return TryGetQualifiedDebouncer(in reportData, out IReportsDebouncer? debouncer)
+                   && debouncer!.Debounce(new ReportMessageFingerprint(new ExceptionFingerprint(message, reportData.Debounce.CallStackHint)), reportData, logType);
+        }
+
         private bool Debounce(in object message, in ReportData reportData, LogType logType)
         {
             if (!debounceEnabled) return false;
 
-            return DebounceInternal(in message, in reportData, logType);
+            if (message is not string stringMessage) // Can't make assumptions about the message type
+                return false;
+
+            return TryGetQualifiedDebouncer(in reportData, out IReportsDebouncer? debouncer)
+                   && debouncer!.Debounce(new ReportMessageFingerprint(stringMessage), reportData, logType);
         }
 
-        protected virtual bool DebounceInternal(in object message, in ReportData reportData, LogType logType)
+        [HideInCallstack]
+        private bool TryGetQualifiedDebouncer(in ReportData reportData, out IReportsDebouncer? debouncer)
```

Comment:
Should a `TryGet` method output a nullable? Isnt it implied that if it couldnt get it, it will return false?

Reply:
It concerns all `TryGet` methods, I am actually not sure, there is no native .NET reference as they use:

`public bool [TryGetValue](https://source.dot.net/System.Private.CoreLib/R/ea6b4e38bc066856.html)(TAlternateKey key, [[MaybeNullWhen](https://source.dot.net/System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/NullableAttributes.cs.html#ad066ae86771e012)(false)] out TValue value)`
and `[MaybeNullWhenAttribute](https://source.dot.net/System.Private.CoreLib/R/4cbcd5c96ef54517.html)` is internal

Reply:
`MaybeNullWhen` is internal